docs/build-from-source.md

```diff
@@ -72,3 +72,56 @@ Production builds optimize code and minimize debug information, while developmen
+
+## Env Variables
+
+To allow for adjustments to be made at compile time, there are several env variables that get replaced during [vite compile](https://vitejs.dev/guide/env-and-mode.html#env-variables).
+These values are then defacto constants, which means code optimizations can remove unused code such as if/else checks using them.
+
+For non-sensitive values, you can add them directly to the appropriate .env file for each mode.
+
+For sensitive values, if building with github actions we recommend using [github secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) and setting them in the [build action environment](https://docs.github.com/en/actions/learn-github-actions/variables).
+You can see an example of how these are used in our [Firebase Live / Preview actions.](https://github.com/jwplayer/ott-web-app/blob/develop/.github/workflows/firebase-live.yml#L14) 
+
+If building manually, you can create an .env.[mode].local file and add the values there. These files are git ignored which will prevent leaking your secrets to version control. 
+
+> Note: env variables must begin with the 'APP_' prefix or they are ignored by our vite configuration.
+
+### APP_DEFAULT_CONFIG_SOURCE
+
+The 8 character ID (or the url path) of the app config from your JWP account that the web app will use to load its content. Be careful to ensure that this config is always available or your app will fail to load.
+
+If you are using pre-compiled builds instead of building the code yourself, you can also set this value with the [defaultConfigSource ini setting](initialization-file.md#defaultconfigsource).
+Keep in mind, if the [defaultConfigSource ini setting](initialization-file.md#defaultconfigsource) is provided, it will be used even if the `APP_DEFAULT_CONFIG_SOURCE` environment variable is set.
+
+### APP_PLAYER_ID
+
+This value determines which player the OTT Web App loads from the JW Platform.
+By default, the OTT Web App uses a global JWP OTT player that has its setting optimized to work properly with the app, but you can change it if you want to use a player directly from your own account.
+
+If you are using pre-compiled builds instead of building the code yourself, you can also set this value with the [playerId ini setting](initialization-file.md#playerid).
+Keep in mind, if the [playerId ini setting](initialization-file.md#playerid) is provided, it will be used even if the `APP_PLAYER_ID` environment variable is set.
+
+> Note: Be careful if using your own player, since some settings in the player can conflict with the way the player is used in the OTT Web App, causing unexpected behavior or UX experiences.
+
+> Note: If you opt to use the default global player, remember to provide your player key via the [APP_PLAYER_LICENSE_KEY](#APP_PLAYER_LICENSE_KEY) env variable or the [playerLicenseKey setting](initialization-file.md#playerLicenseKey) in the ini file.
+
+### APP_PLAYER_LICENSE_KEY
```

Comment:
One more question: will providing analytics token work the same way as with player license key? It can be automatically derived from the app config.

Reply:
Ya, we do have some overlapping data, since the player key also contains the analytics token. The problem with the player key though is that it only applies for web, so having it built into the app makes more sense. Ideally we would just be able to set the analytics token on the player, but that doesn't work.